server.api/src/main/java/org/kaazing/gateway/server/impl/VersionUtils.java

```diff
@@ -151,71 +132,59 @@ public static String getGatewayProductDependencies() {
-        // simplified. Removals and dependencies should no longer be needed.
-
         if (productTitle != null) {
             // We've already run through this before, so do nothing.
             return;
         }
-
         boolean foundJar = false;
-        String artifact = null;
         String[] pathEntries = System.getProperty("java.class.path").split(System.getProperty("path.separator"));
-        HashMap<String, Attributes> products = new HashMap<>(7);
+        Map<String, Attributes> products = new TreeMap<>(Collections.reverseOrder());
         HashSet<String> removals = new HashSet<>(7);
         for (String pathEntry : pathEntries) {
-            if (pathEntry.contains("gateway.server.test") || (pathEntry.contains("gateway.server") && artifact == null)) {
-                artifact = pathEntry;
-                foundJar = true;
+            if (!pathEntry.contains("gateway.server")) {
+                continue;
             }
-        }
-
-        if (artifact == null) {
-            return;
-        }
-
-        try {
-            JarFile jar = new JarFile(artifact);
-            Manifest mf = jar.getManifest();
-            Attributes attrs = mf.getMainAttributes();
-            jar.close();
-            if (attrs == null) {
-                return;
+            if (LOG.isTraceEnabled()) {
+                LOG.trace(format("Found product entry: %s", pathEntry));
             }
-
-            String title = attrs.getValue("Implementation-Title");
-            String version = attrs.getValue("Implementation-Version");
-            String product = attrs.getValue("Kaazing-Product");
-            String dependencies = attrs.getValue("Kaazing-Dependencies");
-            if (product != null && title != null && version != null) {
-                foundJar = true;
-
-                // Store the list of products found, but remove any products
-                // marked as dependencies (i.e. products on which the current
-                // product depends.  We want to find the product that nothing
-                // else depends on.
-                products.put(product != null ? product : title, attrs);
-                if (dependencies != null) {
-                    String[] deps = dependencies.split(",");
-                    Collections.addAll(removals, deps);
+            try (JarFile jar = new JarFile(pathEntry)) {
+                Manifest mf = jar.getManifest();
+                if (mf == null) {
+                    continue;
+                }
+                Attributes attrs = mf.getMainAttributes();
+                if (attrs == null) {
+                    continue;
+                }
+                String title = attrs.getValue("Implementation-Title");
+                String version = attrs.getValue("Implementation-Version");
+                String product = attrs.getValue("Kaazing-Product");
+                String dependencies = attrs.getValue("Kaazing-Dependencies");
```

Comment:
This should be done if only if any of product, title, version is non-null. As such, it should be done somewhere before line 172.